docs/geps/gep-05.rst

```diff
@@ -0,0 +1,124 @@
+=============================================
+GEP 5 — Optional Rounding of Variables
+=============================================
+
++------------+------------------------------------------------------------------+
+| Author     | `Christian Zimpelmann <https://github.com/ChristianZimpelmann>`_ |
++------------+------------------------------------------------------------------+
+| Status     | Draft                                                            |
++------------+------------------------------------------------------------------+
+| Type       | Discussion                                                       |
++------------+------------------------------------------------------------------+
+| Created    | 2022-01-08                                                       |
++------------+------------------------------------------------------------------+
+| Resolution | <url> (required for Accepted or Rejected or Withdrawn)           |
++------------+------------------------------------------------------------------+
+
+
+Abstract
+--------
+
+This GEP describes the implementation of optional rounding of variables in GETTSIM.
+
+
+Motivation and Scope
+--------------------
+
+For several taxes and transfers, German law specifies that these be rounded in specific
+ways. This leads to different use cases.
+
+1. Some applications require the exact, rounded, amount as specified in the law. This is
+   also helpful for creating test cases.
+2. Other applications require smooth functions and the non-rounding error does not
+   matter much.
+
+This document describes how we go about supporting both use cases.
+
+
+Implementation
+--------------
+
+GETTSIM allows for optional rounding of functions' results. Rounding parameters are
+specified in the ``.yaml``-files.
+
+The example below contains GETTSIM's encoding of the fact that the [XXX] law specifies
+that ``grundr_zuschlag_m`` be rounded to the nearest second decimal point.
+
+- The key has to be a function name, in this case ``grundr_zuschlag_m``
+- The parameter ``base`` determines the base to which the variables is rounded. It has
+  to be a floating point number.
+- The parameter ``direction`` has to be one of ``up``, ``down``, or ``nearest``.
+- The reference must contain the reference to the law, which specifies the rounding.
+
+The rounding parameters become part of the dictionary ``policy_params`` once
+``set_up_policy_environment`` is called in the same way as other policy parameters.
+
+.. code-block:: yaml
+
+       rounding:
```

Comment:
Add which file this is and requirements on structure. At which level is rounding specified, for example.